### ISSUE TITLE
[Feature] qwen4_exp: load Niwaki pruned checkpoints (compact expert banks, paired 2-bit PLE)

### ISSUE THREAD

**Issue by cody-htt** [jundot/omlx#3898]
**Is your feature request related to a problem? Please describe.**

[Niwaki](https://niwakiai.com/) publishes pruned Qwen3.8-Flash-Next checkpoints for 64 GB Macs, e.g. [`neopolita/Qwen3.8-Flash-Next-119B-A5B-Niwaki-v2.4-3bit-mlx`](https://huggingface.co/neopolita/Qwen3.8-Flash-Next-119B-A5B-Niwaki-v2.4-3bit-mlx). They keep `model_type: qwen4_exp`, but a `niwaki` section in `config.json` declares a different tensor layout:

- `kept_experts`: some layers store only a compact bank of experts (e.g. 16 of 512). Routing is still top-k over all 512; experts that aren't stored contribute 0.
- `shared_only_layers`: the routed bank is removed on some layers.
- `maps_unfolded`: a trained output map is applied to each MoE block.
- `niwaki_ple_pair`: PLE shards are stored 2-bit in a paired-row layout.

oMLX discovers these checkpoints but fails to load them:

```
Expected shape (2500012, 10) but received shape (625003, 40) for parameter
language_model.model.layers.1.ple.ple_embedding.ngram_embedding.shards.0.weight
```

**Describe the solution you'd like**

I'd like to contribute a PR, but wanted to check the approach first.

- A new `omlx/patches/qwen4_exp_niwaki/` package, applied only when `config.json` has a `niwaki` section. It's original Apache-2.0 code, and no vendored files change.
- A per-load scope, like `_force_qwen4_exp_sanitize_on_load`, swaps in the Niwaki module types before strict loading (subclasses of `Qwen3_5MoeSparseMoeBlock` plus a paired PLE shard), then restores the original loader.
- The verify path has to go through the Niwaki block's own `__call__`. Today there are two sites that assume all 512 experts are addressable: `qwen35_moe_router`'s verify wrap, and, when that wrap isn't installed, mlx-vlm's unwrapped `_feed_forward`. Both re-derive routing over 512 experts and index past the compact bank. Stock mlx-vlm doesn't error on this, it returns all-NaN logits, which is the harder failure to notice. Two fixes, applied so the outcome doesn't depend on patch order:
  - the router wrap skips Niwaki blocks;
  - a small `isinstance`-gated delegation on `_feed_forward` forwards to the block's own `__call__`.
- For Niwaki checkpoints only, PLE always stays resident, because the SSD/mmap PLE path can't read the paired layout. Expert offload is rejected with a clear error.
- Tests use the tiny `qwen4_exp` fixture for each layout, plus a short docs note.

Performance tuning would be a separate follow-up PR.

**Describe alternatives you've considered**

- Editing the vendored model code: that would make pin bumps harder.
- Adding support to mlx-vlm first: oMLX would still need glue code for its vendored model.

**Additional context**

**Functional check.** Spike on v0.7.0.dev4 (`14194fe`) with a throwaway load hook. The 119B loads and serves correctly, and passes determinism at 8K (3/3 replays + tool call at temperature 0), a 6-turn agent loop with prefix-cache restores, and concurrent requests.

**Perf comparison.** Same Mac (M5 Max, 64 GB), same prompts, thinking on, temperature 0, 2,048 output tokens. Baseline is stock `mlx_vlm.server` 0.7.1 running the model repo's own loader.

| Prompt | oMLX dev4 prefill / TG | mlx-vlm prefill / TG |
|---|---|---|
| 1K | 1,071 / 43.0 tok/s | 1,174 / 42.0 tok/s |
| 2K | 1,083 / 43.1 | 966 / 42.2 |
| 4K | 940 / 40.7 | 1,128 / 41.6 |
| 8K | 728 / 37.4 (TTFT 12.0 s) | 1,208 / 40.5 (TTFT 7.2 s) |

- **Generation speed:** on par.
- **Prefill:** oMLX falls behind above 2K, but most of that gap is memory headroom, not the Niwaki code. At about 49 GB in use, the prefill memory guard throttled chunked prefill down to 32-token chunks, which is where the throughput goes on this model size.
- **Memory limit:** with macOS's default Metal limit (51.8 GB), the guard rejects prompts of about 6K tokens or more (`prefill_memory_exceeded`) for this 45 GB model. It worked after raising `iogpu.wired_limit_mb` to 59392. The docs note would mention that setting.
- **Remaining costs from Niwaki itself:** the masked layers bypass the fused router, and the paired PLE isn't fused. Those are the follow-up.

I'll re-validate on `main` (mlx-vlm `ea79808`). The PR will have a real-model check on the 119B; `shared_only_layers` and `maps_unfolded` will be tested synthetically only.

Questions:

1. Would you accept this in-tree?
2. Is a separate patch package the right place, or should it go inside `mlx_vlm_qwen4_exp_compat`?
3. Is it OK for the first PR to keep PLE resident and reject expert offload for these checkpoints?

Related: #3170
